### DISCUSSION TITLE
ISolar-SMG-II-4KW-24V WiFi

### DISCUSSION THREAD

**Discussion by trianglesis** [softwarecrash/Solar2MQTT-ESP8266#112] [General]
Hello,
First of all, thanks for your work.

I'm setting up my inverter using your guide. The only problem is that this invertor model is too modern. 
I can't check what protocol it uses, and Solar2MQTT is based on "ESP8266, WeMos D1 mini" (clone or something) is blinking two times.

Is there some workaround for capturing inverter data for debugging or reverse engineering? 
I saw that many older EASun inverters are supported, and there may be a small detail that I'm not aware of. 

- ![image](https://github.com/user-attachments/assets/52174376-e6ec-49a0-b9b9-f5e4388336e0)

- ![IMG_20240906_224638921](https://github.com/user-attachments/assets/4aed8940-6173-4dd2-943d-7f79dda73e11)

The product:

1. Amazon: https://www.amazon.de/-/en/POWLAND-Inverter-Off-Grid-Maximum-Batteries/dp/B0D8FH6Q88/
2. Manufacturer: https://www.easun-energy.com/products/easun-power-4kw-solar-inverter-100a-mppt-solar-charge-controller-with-wifi

**Comment by softwarecrash** [comment]
the two  blinks  means  no device,, so your device with the wifi dingle looks  like many others, on this wifi serial interface you can connect the esp.
the only way to capture data is to get in the serial communikation between the wifi dongle and the inverter

**Reply by trianglesis** [reply]
I couldn't get the thing to work, so I moved to another project, which worked. 
You can close this issue now.

The project that worked: /syssi/esphome-smg-ii

Thanks!

**Comment by Haso-M** [comment]
Isolar SMG II & SMG III are on ModbusRTU. Not compatible.

**Reply by garremmash** [reply]
Could you please tell me if this inverter is supported? It seems like Modbus was added in the firmware, or is that something else?

**Reply by softwarecrash** [reply]
modbus is only the comunication, its on testing now, we need the protocol sheet for your device

**Reply by Haso-M** [reply]
You can find Modbus register map on "docs" folder of /syssi/esphome-smg-ii. I personally tested syssi's project on EASUN SMG-III and works. But ESPHome is too restrictive for my taste.

**Reply by softwarecrash** [reply]
ok, i work on serveral bugs bevor i can test modbus and implement new things, this wil take a while

**Reply by Sensorv** [reply]
Any news about support for this inverter (**ISolar-SMG-II-4KW-24V WiFi**)?
When will it be possible to use it with solar2mqtt?
Thank you

**Reply by softwarecrash** [reply]
@bigdigital have you a idea it will work with the new registers?
